includes/class-wc-payments-payment-method-messaging-element.php

```diff
@@ -146,6 +146,8 @@ public function init() {
 
 		if ( $product ) {
 			$script_data['isBnplAvailable'] = WC_Payments_Utils::is_any_bnpl_method_available( array_values( $bnpl_payment_methods ), $country, $currency_code, $product_price );
+		} else {
+			$script_data['isBnplAvailable'] = WC_Payments_Utils::is_any_bnpl_method_available( array_values( $bnpl_payment_methods ), $country, $currency_code );
 		}
```

Comment:
I still don't understand why we have two scenarios for `is_any_bnpl_method_available` here. I assume that we don't send an amount when we are in the cart page since the totals could change, but it's the same case for variable products.